includes/reader-revenue/class-woocommerce-connection.php

```diff
@@ -80,16 +80,26 @@ public static function set_up_membership( $email_address, $full_name, $frequency
 			}
 		}
 		if ( $should_create_account ) {
-			Logger::log( 'This order will result in a membership, creating account for user.' );
-			$user_login = sanitize_title( $full_name );
-			$user_id    = wc_create_new_customer( $email_address, $user_login, '', [ 'display_name' => $full_name ] );
-			if ( is_wp_error( $user_id ) ) {
-				return $user_id;
+			if ( Reader_Activation::is_enabled() ) {
+				$user_id = Reader_Activation::register_reader( $email_address, $full_name );
+				if ( is_wp_error( $user_id ) ) {
+					return $user_id;
+				}
+				if ( ! absint( $user_id ) ) {
+					$user_id = null;
+				}
+			} else {
+				Logger::log( 'This order will result in a membership, creating account for user.' );
+				$user_login = sanitize_title( $full_name );
+				$user_id    = wc_create_new_customer( $email_address, $user_login, '', [ 'display_name' => $full_name ] );
```

Comment:
I wouldn't call duplicated, the original intended behavior is preserved while reader activation is experimental.

WP core already provides a straightforward API for generic user creation that should be used if not making use of reader activation logic. I don't see why reader activation would provide methods outside of its concept.

Reply:
I was thinking about a method that would either `wp_insert_user` or `wc_create_new_customer` – but on a second thought it should not be used here because at this point we only wanna handle customer creation, not user creation. I was overthinking it. 

Still, I think it'd be more readable if `Reader_Activation` - in this code block - only set metadata on the created user, not take over the user creation and flow. Something like:

```php
$user_id = wc_create_new_customer …
if ( Reader_Activation::is_enabled() ) {
  Reader_Activation::setup_reader( $user_id );
}
wp_set_current_user …
```

But it's not a blocker.

Reply:
A "reader" is a fairly flexible concept:

https://github.com/Automattic/newspack-plugin/blob/feat/reader-activation-donate-block/includes/class-reader-activation.php#L131-L154

Currently any subscriber or customer is eligible to use the reader-related tools, including `register_reader( $email )`. What happens when we call `setup_reader( $user_id )`?

Reply:
To clarify on why we have the user meta to also determine the same thing: the user roles that determine a reader are filterable. If by any chance this is filtered to an empty array, we must still be able to know readers that were created through this flow.

Reply:
That's an interesting idea, we should probably do it in a separate method to not necessarily trigger all the side effects of a reader registration if we just want to set the user as a reader.

If you don't mind, I think we could work on that on a separate PR, once we have more practical use on other cases.